src/DotVVM.Framework/Resources/Scripts/metadata/metadataHelper.ts

```diff
@@ -0,0 +1,40 @@
+﻿import { getTypeInfo } from "./typeMap";
+
+export function getTypeId(viewModel: any): string {
```

Comment:
We should either return `string | undefined` or check that the $type property actually exists. It just may not be there for some objects (for example if it comes from API bindings).

Reply:
I prefer the variant with `string | undefined`